modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationTaskRunner.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.calc.runner;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+import com.opengamma.strata.basics.CalculationTarget;
+import com.opengamma.strata.calc.CalculationRules;
+import com.opengamma.strata.calc.CalculationRunner;
+import com.opengamma.strata.calc.Column;
+import com.opengamma.strata.calc.marketdata.CalculationEnvironment;
+
+/**
+ * Component that provides the ability to run calculation tasks.
+ * <p>
+ * This interface is the lower-level counterpart to {@link CalculationRunner}.
+ * It provides the ability to calculate results based on {@link CalculationTasks}.
+ * Unless you need to optimize, the {@code CalculationRunner} is a simpler entry point.
+ * <p>
+ * Each {@code CalculationTasks} instance contains one {@code CalculationTask} for each
```

Comment:
It might be clearer to describe a calculation task as corresponding to a single cell in a grid where the rows are defined by targets and the columns by measures.